lib/spack/spack/util/executable.py

```diff
@@ -213,7 +215,7 @@ def streamify(arg, mode):
                         sys.stderr.write(errstr)
 
             rc = self.returncode = proc.returncode
-            if fail_on_error and rc != 0 and (rc not in ignore_errors):
+            if rc != 0 and (rc not in ignore_errors):
```

Comment:
```suggestion
            if rc != 0:
```
I think that Executable.error should be set regardless of whether we ignore the error, and push this check down to where we determine whether to raise.